iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnBuffers.cpp

```diff
@@ -1085,14 +1087,37 @@ struct TensorReshapeOpConversion
       linalg::TensorReshapeOp reshapeOp, ArrayRef<Value> operands,
       ConversionPatternRewriter &rewriter) const override {
     linalg::TensorReshapeOp::Adaptor adaptor(operands);
-    if (Value buffer =
-            resolveResult(reshapeOp.src(), adaptor.src(), reshapeOp.result(),
-                          resultTensorToBufferMap))
-      rewriter.replaceOp(reshapeOp, buffer);
-    else
+    // If result has an associated buffer.
+    Value bufferForResult = resultTensorToBufferMap.lookup(reshapeOp.result());
+    if (!bufferForResult) {
+      // This is not a reshape before store_tensor. Replace this op with a
+      // reshape on buffers.
       rewriter.replaceOpWithNewOp<linalg::ReshapeOp>(
           reshapeOp, getMemrefTypeForTensor(reshapeOp.result()), adaptor.src(),
           reshapeOp.reassociation());
+      return success();
+    }
+
+    // Look at all uses of bufferForResult in reshape ops. If once of those is
+    // the input operand, there is nothing to do.
+    bool insertCopy = true;
+    for (auto &uses : bufferForResult.getUses()) {
+      if (auto bufferReshapeOp = dyn_cast<linalg::ReshapeOp>(uses.getOwner())) {
+        if (bufferReshapeOp && bufferReshapeOp.result() == adaptor.src()) {
+          insertCopy = false;
+          break;
+        }
+      }
+    }
+
```

Comment:
How about:

```
     bool insertCopy = llvm::all_of(bufferForResult.getUses(), [&](auto &use) {
       if (auto reshapeOp = dyn_cast<linalg::ReshapeOp>(use.getOwner())) {
          return reshapeOp.result() != adaptor.src();
       }
       return true;
     });
```

This looks cleaner to me, and it also match the comment directly?

Reply:
That seems right. Changed it, but made it !llvm::any_of. Easier for me to parse.